tests/test_utils.py

```diff
@@ -345,3 +357,76 @@ def test_msdf_from_mappings(self):
         self.assertEqual(1, len(new_msdf.df.index))
         self.assertEqual(rows[0], tuple(msdf.df.iloc[0]))
         self.assertEqual(new_msdf.metadata, msdf.metadata)
+
+    def test_get_dict_from_mapping(self):
```

Comment:
this tests many many different things - it's not helpful if we ever need to debug or understand what happened. Instead, it's preferable to make many small, minimal tests for each part of the function that's being tested